classes/assessment/assign.php

```diff
@@ -53,4 +66,168 @@ public function get_start_date(): ?int {
     public function get_end_date(): ?int {
         return $this->sourceinstance->duedate;
     }
+
+    /**
+     * Apply EC extension to the assessment.
+     *
+     * @param ec $ec The EC extension.
+     * @return void
+     */
+    protected function apply_ec_extension(ec $ec): void {
```

Comment:
just noting that if we need to implement this per activity, the other pull requests allowing push would only be providing partial support and would need follow up story to implement this in each

Reply:
Yes, that's correct, and the `apply_sora_extension()` function too.